### PR TITLE
SAK-47769 - Open Lessons LTI new window links with JavaScript

### PR DESCRIPTION
**Pull request by csev**

### CODE REVIEW

File: lessonbuilder/tool/src/webapp/js/show-page.js [hunk 1]
```diff
@@ -2974,6 +2974,16 @@ $(document).ready(function() {
 	$('.no-highlight').folderListing({
 		enableHighlight: false,
 	});
+
+	$('a[href*="access/basiclti"]').each(function() {
+		var href = this.href;
+		if ( this.href == null ) return;
```
Comment (bjones86) [inline lessonbuilder/tool/src/webapp/js/show-page.js:2980]: You should use `===` here, unless you're relying on type coercion.

```suggestion
		if ( this.href === null ) return;
```

File: lessonbuilder/tool/src/webapp/js/show-page.js [hunk 2]
```diff
@@ -2974,6 +2974,16 @@ $(document).ready(function() {
...
+		if ( this.href.indexOf('/access/basiclti') < 0 ) return
+		if ( '_blank' != this.target ) return
```
Comment (bjones86) [inline lessonbuilder/tool/src/webapp/js/show-page.js:2982]: Same thing here: 
```suggestion
		if ( '_blank' !== this.target ) return
```

File: lessonbuilder/tool/src/webapp/js/show-page.js [hunk 3]
```diff
@@ -2974,6 +2974,16 @@ $(document).ready(function() {
...
+		if ( this.onclick != null ) return;
```
Comment (bjones86) [inline lessonbuilder/tool/src/webapp/js/show-page.js:2983]: And here:
```suggestion
		if ( this.onclick !== null ) return;
```

**Review by bjones86 — CHANGES_REQUESTED** [review]
Some minor suggestions. https://www.freecodecamp.org/news/js-type-coercion-explained-27ba3d9a2839/

**Comment by csev** [comment]
@bjones86 I accepted your suggestions.  My personal rule in these cases is to use == and != unless I am expecting that type coercion will potentially cause a problem (i.e. like an integer 0 turning into null or false) - unless I know how it will fail, I use the more common form and the triple equals for to call attention to the danger that type coercion might cause.  But in this case the triple form is OK and slightly faster methinks.   The say that a good FORTRAN programmer can write FORTRAN in any language - and since FORTRAN does not have triple equals - I tend to use it sparingly in my Python / JS / PHP code :) In some ways I semantically prefer the "is" operator in Python to make sure to call attention to the fact that its semantics are quite different than ==.  But thanks for the comments - you are correct.

**Review by bjones86 — APPROVED** [review]

File: lessonbuilder/tool/src/webapp/js/show-page.js [hunk 4]
```diff
@@ -2974,6 +2974,16 @@ $(document).ready(function() {
 	$('.no-highlight').folderListing({
 		enableHighlight: false,
 	});
+
+	$('a[href*="access/basiclti"]').each(function() {
+		var href = this.href;
```
Comment (bjones86) [inline lessonbuilder/tool/src/webapp/js/show-page.js:2979]: One more small suggestion:
```suggestion
		const href = this.href;
```

**Pull request merged**